src/main/java/htsjdk/samtools/util/AsciiWriter.java

```diff
@@ -70,17 +71,10 @@ public void flush() throws IOException {
      * All other Writer methods vector through this, so this is the only one that must be overridden.
      */
     @Override
-    public void write(final char[] chars, int offset, int length) throws IOException {
-        while (length > 0) {
-            final int charsToConvert = Math.min(length, buffer.length - numBytes);
-            StringUtil.charsToBytes(chars, offset, charsToConvert, buffer, numBytes);
-            numBytes += charsToConvert;
-            offset += charsToConvert;
-            length -= charsToConvert;
-            if (numBytes == buffer.length) {
-                os.write(buffer, 0, numBytes);
-                numBytes = 0;
-            }
-        }
+    public void write(final char[] chars, int off, int len) throws IOException {
+        String str = new String(chars,off,len);
+        byte[] b = str.getBytes(StandardCharsets.UTF_8);
```

Comment:
This is going to write all SAM content as UTF-8, but I think the spec only permits UTF-8 in a few places, and requires 7 bit ascii elsewhere. So we may need a different approach..

Reply:
It seems like we're actually just doing a costly reimplementation of a standard Java Writer here now.  It seems like validation of the character ranges should be done in the SamRecord and not here.   Maybe this class has outlived it's usefulness and we can just use a standard Writer in place of it.